### PR TITLE
Some performance reduce-mixin fixes

### PR DESCRIPTION
**Pull request by JKL98ISR**
related to #1550

### CODE REVIEW

**Review by ItayGabbay — DISMISSED** [review]

File: deepchecks/tabular/checks/model_evaluation/performance_report.py [hunk 1]
```diff
@@ -150,6 +150,7 @@ def reduce_output(self, check_result: CheckResult) -> Dict[str, float]:
         """Return the values of the metrics for the test dataset in {metric: value} format."""
         df = check_result.value
         df = df[df['Dataset'] == DatasetKind.TEST.value]
+        df = df.groupby('Metric').mean().reset_index()
```
Comment (matanper) [inline deepchecks/tabular/checks/model_evaluation/performance_report.py:153]: I don't think this makes sense to do a mean

Comment (JKL98ISR) [reply deepchecks/tabular/checks/model_evaluation/performance_report.py:153]: we can't start making assumptions about params all need to be able to be passed as options...
half of our params coming from unknown sources like the alternative metrics

Comment (JKL98ISR) [reply deepchecks/tabular/checks/model_evaluation/performance_report.py:153]: the default sklearn metrics behavior for metrics if you want a single value is mean.
"Calculate metrics for each label, and find their unweighted mean. This does not take label imbalance into account."

Comment (matanper) [reply deepchecks/tabular/checks/model_evaluation/performance_report.py:153]: after talking with @noamzbr the main points are:
1. If someone wants an averaged metric (not per-class), then he should use averaged scorers.
2. Using per-class metrics is legit for other use cases, like monitoring the worst-performing class. Therefore in order to make it possible we can add `per_class_reduce_fn` which accepts a string of pre-defined values (min, max, median, std, etc...), and then in the per-class case, we apply this reduce

Comment (JKL98ISR) [reply deepchecks/tabular/checks/model_evaluation/performance_report.py:153]: we need the reduced_output to work in every case of the check
also i commented on why the reduce function should be as the check sol did

Comment (JKL98ISR) [reply deepchecks/tabular/checks/model_evaluation/performance_report.py:153]: how is the check's default behavior misconfigured?

> where? you posted a sentence from unknown source and context
what do you mean? the user should be able to give a reduce function as it is in sol's check

File: deepchecks/tabular/checks/model_evaluation/performance_report.py [hunk 2]
```diff
@@ -150,6 +156,7 @@ def reduce_output(self, check_result: CheckResult) -> Dict[str, float]:
         """Return the values of the metrics for the test dataset in {metric: value} format."""
         df = check_result.value
         df = df[df['Dataset'] == DatasetKind.TEST.value]
+        df = df.groupby('Metric').aggregate(self.reduce).reset_index()
```
Comment (matanper) [inline deepchecks/tabular/checks/model_evaluation/performance_report.py:159]: the aggregate function accepts strings and functions?

Comment (JKL98ISR) [reply deepchecks/tabular/checks/model_evaluation/performance_report.py:159]: yas, even tested

**Review by matanper — APPROVED** [review]

**Pull request merged**